comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -164,6 +179,14 @@ func processCacheDifference(procCacheA map[int32]*procutil.Process, procCacheB m
 	return newProcs
 }
 
+func (c *collector) detectLanguages(processes []*procutil.Process) []*languagemodels.Language {
+	languageInterfaceProcs := make([]languagemodels.Process, len(processes))
+	for i, proc := range processes {
+		languageInterfaceProcs[i] = languagemodels.Process(proc)
+	}
+	return languagedetection.DetectLanguage(languageInterfaceProcs, c.systemProbeConfig)
+}
```

Comment:
Forgot to mention, we should only collect the languages if the config `language_detection.enabled` is true.